src/math/LinearCombinationKernel.cc

```diff
@@ -107,6 +107,24 @@ std::shared_ptr<Kernel> LinearCombinationKernel::clone() const {
     return retPtr;
 }
 
+std::shared_ptr<Kernel> LinearCombinationKernel::resized(int width, int height) const {
+    KernelList kernelList;
+
+    for (KernelList::const_iterator kIter = _kernelList.begin(), kEnd = _kernelList.end(); kIter != kEnd;
+         ++kIter) {
+        std::shared_ptr<Kernel> basisKernelPtr = (*kIter)->resized(width, height);
+        kernelList.push_back(basisKernelPtr);
```

Comment:
Takes more characters and is harder to read (in my opinion) than

```
    for (const std::shared_ptr<Kernel>& kIter : getKernelList()) {
        kernelList.push_back(kIter->resized(width, height));
    }
```
Performance gains worth it?